src/components/calendars/large/CalendarView.tsx

```diff
@@ -1,30 +1,41 @@
-import React from 'react';
+import React, { useCallback, useMemo } from 'react';
 
 import styled from '@emotion/styled';
 
-import CalendarCell from '@/components/calendars/large/CalendarCell';
-import CalendarPlanLayer from '@/components/calendars/large/CalendarPlanLayer';
-import usePlanDrag, { MouseEventHandler } from '@/hooks/usePlanDrag';
+import CalendarLayer from './CalendarLayer';
+import CalendarOverlay from './CalendarOverlay';
+import CalendarWeek from './CalendarWeek';
+import usePlanDrag from '@/hooks/usePlanDrag';
+import Plan from '@/plan/Plan';
 import useDateState from '@/stores/date';
 import useFocusedPlanState from '@/stores/plan/focusedPlan';
-import { getStartAndEndDateInMonth } from '@/utils/calendar/dayHandler';
 import { getCalendarInfo } from '@/utils/calendar/getCalendarInfo';
-import { getCalendarPlans } from '@/utils/calendar/getCalendarPlans';
 import { dummy } from '@/utils/plan/dummy';
+import { getDaysPlanManager } from '@/utils/plan/getDaysPlanManager';
 
 const CalendarView = () => {
-  const { focusedPlan, createDragPlan, isDragging } = useFocusedPlanState();
-  const { onChangeStoreDate, year, month, day } = useDateState();
-  const { changeCurrentDate, currentDateRef, onMouseMove } = usePlanDrag();
+  const { year, month, day } = useDateState();
+  const { focusedPlan, isDragging, createDragPlan } = useFocusedPlanState();
+  const { currentDateRef, onMouseMove, changeCurrentDate } = usePlanDrag();
 
-  const calendarInfos = getCalendarInfo({ year, month, day });
-  const dates = getStartAndEndDateInMonth(calendarInfos);
-  const selectedPlanArgs = focusedPlan ? [focusedPlan] : [];
+  const calendarInfos = useMemo(
+    () => getCalendarInfo({ year, month, day }),
+    [year, month, day],
+  );
+
+  const planManagers = useMemo(
+    () => getDaysPlanManager(dummy, calendarInfos),
+    [dummy, calendarInfos],
+  );
 
-  const calendarPlanViews = getCalendarPlans(dummy, ...dates);
-  const selectedPlanViews = getCalendarPlans(selectedPlanArgs, ...dates);
+  const focusedPlanArgs = focusedPlan ? [new Plan(focusedPlan)] : [];
+
+  const focusedPlanManager = useMemo(
+    () => getDaysPlanManager(focusedPlanArgs, calendarInfos),
+    [focusedPlanArgs, calendarInfos],
```

Comment:
`focusedPlanArgs` 의 경우 리렌더링시 매번 새로운 배열을 가지는 변수인 것 같습니다. 그렇다면 deps 배열에 `focusedPlanArgs`를 넣어주는 것은 `useMemo`의 이점을 살리지 못하는 것 같습니다.

```suggestion
 const focusedPlanManager = useMemo(
    () => getDaysPlanManager(focusedPlan ? [new Plan(focusedPlan)] : [], calendarInfos),
    [focusedPlan, calendarInfos],
)
```

로 해야 `useMemo`의 이점을 살릴 수 있을 것 같습니다.

Reply:
확인했습니다. 렌더링 상황에서 `focusedPlanArgs`가 새로운 주소값을 가지는 것 같습니다. 기존 코드에서 조금씩 수정하다보니 이런 실수를 했네요..